### PR TITLE
WIP Create mosaic class version

### PR DESCRIPTION
**Pull request by kylebarron**

### CODE REVIEW

File: cogeo_mosaic/model.py [hunk 1]
```diff
@@ -24,3 +31,169 @@ class MosaicJSON(BaseModel):
...
+                    sort_cover=tile_cover_sort,
+                )
+
+            if dataset and maximum_items_per_tile:
+                dataset = dataset[0:maximum_items_per_tile]
+
+            if dataset:
+                mosaic_definition["tiles"][quadkey] = [f["identifier"] for f in dataset]
+
+        return MosaicJSON.from_dict(mosaic_definition)
```
Comment (vincentsarago) [inline cogeo_mosaic/model.py:199]: I think going directly with `return cls(**mosaic_definition)` is fine here

File: cogeo_mosaic/model.py [hunk 2]
```diff
@@ -24,3 +31,169 @@ class MosaicJSON(BaseModel):
     bounds: List[float] = Field([-180, -90, 180, 90])
     center: Optional[Tuple[float, float, int]]
     tiles: Dict[str, List[str]]
+
+    @classmethod
+    def from_dict(cls, mosaic_def: Dict):
+        return MosaicJSON(**dict(mosaic_def))
```
Comment (vincentsarago) [inline cogeo_mosaic/model.py:37]: if it's already a dict you don't need `dict(mosaic_def)`?

File: cogeo_mosaic/model.py [hunk 3]
```diff
@@ -24,3 +31,169 @@ class MosaicJSON(BaseModel):
...
+        datatype = {feat["properties"]["datatype"] for feat in features}
+        if len(datatype) > 1:
+            raise Exception("Dataset should have the same data type")
+
+        return MosaicJSON.from_features(
+            features, minzoom=minzoom, maxzoom=maxzoom, quiet=quiet, **kwargs
+        )
+
+    @classmethod
+    def _create_mosaic(
```
Comment (vincentsarago) [inline cogeo_mosaic/model.py:101]: I think this could be `from_feature` directly.

File: cogeo_mosaic/model.py (hunk 1 above)
Comment (kylebarron) [reply cogeo_mosaic/model.py:199]: I was looking at the Geopandas source, and I got the `Classname(object)` syntax from them: https://github.com/geopandas/geopandas/blob/f13dafb881eaaf2bbf296e6f12189cc2c21a4216/geopandas/geodataframe.py#L387

File: cogeo_mosaic/model.py (hunk 2 above)
Comment (kylebarron) [reply cogeo_mosaic/model.py:37]: I know, but it doesn't hurt to coerce

File: cogeo_mosaic/model.py (hunk 3 above)
Comment (kylebarron) [reply cogeo_mosaic/model.py:101]: I feel that it's better to split it up for composability. It's a nice logical separation

File: cogeo_mosaic/model.py (hunk 2 above)
Comment (vincentsarago) [reply cogeo_mosaic/model.py]: well it just doesn't make sense to me to re-dict a dict ;-)

File: cogeo_mosaic/model.py (hunk 1 above)
Comment (geospatial-jeff) [reply cogeo_mosaic/model.py:199]: I prefer the `cls` constructor.  Regardless you don't need `from_dict`.

Comment (kylebarron) [reply cogeo_mosaic/model.py:199]: Well it isn't _wrong_ at least, since it works

**Pull request closed without merging**